src/nvim/event/stream.c

```diff
@@ -90,6 +90,13 @@ void stream_close(Stream *stream, stream_close_cb on_stream_close, void *data)
   }
 }
 
+void stream_may_close(Stream *stream)
```

Comment:
"try" would be more appropriate: `stream_try_close`

("maybe" usually describes the state of a thing, rather than the intent of a function)